### PR TITLE
Add initial ELO calculator implementation

### PR DESCRIPTION
**Pull request by ra-97**

### CODE REVIEW

File: app/Models/Game.php [hunk 1]
```diff
@@ -32,4 +34,58 @@ public function getHasWinnerAttribute(): bool
...
+     * Calculates and updates userA and userB's ELO rating.
+     */
+    public function UpdateUserElo()
+    {
+        // Currently this function accepts only two users as arguments and a winning
+        // indication factor. This function can be extended to accept more than two
+        // and handle the deciding which player won and which lost.
+
+        // If there is no winner, do nothing
+        if (!$this->getHasWinnerAttribute()) {
```
Comment (realpoke) [inline app/Models/Game.php:56]: Just do `$this->hasWinner` since it's an attribute.

File: app/Models/Game.php [hunk 2]
```diff
@@ -32,4 +34,58 @@ public function getHasWinnerAttribute(): bool
...
+            return;
+        }
+
+        $winner = $this->winner();
```
Comment (realpoke) [inline app/Models/Game.php:60]: Only using the winner once, so let's not make it a variable.

File: app/Models/Game.php [hunk 3]
```diff
@@ -32,4 +34,58 @@ public function getHasWinnerAttribute(): bool
...
+        $userA = null;
+        $userB = null;
+
+        foreach ($users as $user) {
+            if ($userA === null) {
+                $userA = $user;
+            } else {
+                $userB = $user;
+            }
+        }
```
Comment (realpoke) [inline app/Models/Game.php:72]: The `$users` is a [collection](https://laravel.com/docs/10.x/collections#main-content) so you can do something like `$userA->first()` and `$userB->last()` maybe second is a thing too.

**Review by realpoke — CHANGES_REQUESTED** [review]
Great, just small tweaks and the hasWinner used a bit wrong! Great job!

File: app/Models/Game.php (hunk 3 above)
Comment (ra-97) [reply app/Models/Game.php:72]: Seems like there is not second() member, but I think we could do a skip() and then first() again, but I think for now last() looks cleaner.

**Review by realpoke — APPROVED** [review]

**Pull request merged**